### DISCUSSION TITLE
how to run segmentation in batch mode

### DISCUSSION THREAD

**Discussion by ML33M** [SchmollerLab/Cell_ACDC#518] [Q&A]
Hi guys, thank you so much for making this tool! I was in the process of pulling cellpose down to connect to our local HPC. In the same vein, if I have lots of images sitting on a server, how can I run Cell_ACDC segmentation module in on our local computing cluster so I can batch the large amount of images, and ask Cell_ACDC to do the segementation and export the segmentation masks.

**Comment by ElpadoCan** [comment]
Hi @ML33M, 

Are the images on the server already organised with the folder structure required by Cell-ACDC? 

If not, we need to address that first. 

If yes, you can set up a segmentation and tracking workflow locally on some sample images using module 2 (which you can run by pressing the `2. Launch segmentation module... ` button on the Cell-ACDC launcher) and at the end of the wizard instead of running the process you choose to save the workflow to a configuration file. Then you just open the configuration file in a text editor app and modify the paths to the experiment folders you want to analyse (these will be the paths of the images on the HPC server).  Finally, you copy the configuration file to the server, open the terminal and run the workflow with the command `acdc -p <path_to_configuration_file>`. Of course you will need to have Cell-ACDC installed on the server too (I'm guessing on the server you don't need the GUI library which is default installation). 

Let me know if it's clear and if it works, thanks! 

Best,
Francesco

**Reply by ML33M** [reply]
Thank you for this. I definately get ahead of myself when asking the question. I do not have it on my work computer yet, I will ask the IT for approval then test this out, definitely keep you updated. I suspect somewhere there will be some dependency issues, pending on what version of python our cluster has...

**Reply by ElpadoCan** [reply]
Absolutely no problem, should you have installation issues and you cannot solve them feel free to report them and I'm certain we can fix them. Good luck!
